### PR TITLE
allow arbitrary keys to be nested under the api object

### PR DESCRIPTION
**Pull request by ryanwilliamquinn**

### CODE REVIEW

File: lib/schemas/api.js [hunk 1]
```diff
@@ -1,9 +1,3 @@
 const joi = require('joi')
 
-module.exports = joi.object().keys({
-  controllers: joi.object().required(),
-  models: joi.object().required(),
-  policies: joi.object().required(),
-  services: joi.object().required(),
-  config: joi.object().optional()
-})
+module.exports = joi.object().required()
```
Comment (jaumard) [inline lib/schemas/api.js]: controllers, models, policies, services and config are required by Trails and should stay like this, if you want to allow more keys take a look at https://github.com/hapijs/joi/blob/v9.0.0-5/API.md#objectunknownallow to allow any keys but required the ones we need

Comment (ryanwilliamquinn) [inline lib/schemas/api.js]: Sounds good, updated the pr. Thanks for the review.

**Comment by tjwebb** [comment]
@jaumard is this able to be merged?

File: lib/schemas/api.js (hunk 1 above)
Comment (jaumard) [inline lib/schemas/api.js]: For me it should look like : 

```
module.exports = joi.object().keys({
  controllers: joi.object().required(),
  models: joi.object().required(),
  policies: joi.object().required(),
  services: joi.object().required(),
  config: joi.object().optional()
}).unknow()
```

It force to have correct API structure and allow extra keys :)

**Comment by jaumard** [comment]
@tjwebb almost, need to set validation schema too 

```
module.exports = joi.object().keys({
  controllers: joi.object().required(),
  models: joi.object().required(),
  policies: joi.object().required(),
  services: joi.object().required(),
  config: joi.object().optional()
}).unknow()
```

And I think we're good to go :)

**Pull request merged**